### ISSUE TITLE
Autoload of mixins

### ISSUE THREAD

**Issue by zerkms** [vimeo/psalm#3174]
The core of this report lies in the https://github.com/webmozart/assert/pull/126

I have discovered that types referred by `@mixin` annotation are not loaded automatically by psalm.

How to reproduce:

Foo.php:
```php
<?php
declare(strict_types=1);

namespace Zerkms;

/**
 * @mixin Mixin
 *
 * @method static void nullOrBoolean($value, $message = '')
 */
final class Foo
{
}
```

Mixin.php
```php
<?php

namespace Zerkms;

interface Mixin
{
    /**
     * @psalm-pure
     * @psalm-assert null|bool $value
     * @param mixed $value
     * @param string $message
     * @throws \InvalidArgumentException
     */
    static function nullOrBoolean($value, $message = '');
}
```

code.php:
```php
<?php

use Zerkms\Foo;

function giveMeString(?bool $s): void {}

/** @var mixed $v */
$v = 'foo';

Foo::nullOrBoolean($v);

giveMeString($v);
```

Unless there is an explicit `require`/`include` in the `Foo.php` - psalm does not load the corresponding `@mixin` so fails with

```
ERROR: UndefinedMethod - src/code.php:10:1 - Method Zerkms\Foo::nullorboolean does not exist (see https://psalm.dev/022)
Foo::nullOrBoolean($v);


ERROR: MixedArgument - src/code.php:12:14 - Argument 1 of giveMeString cannot be mixed, expecting bool|null (see https://psalm.dev/030)
giveMeString($v);
```

While if you just add `require __DIR__ . '/Mixin.php';` it would type check.

Not sure if this is intended and deliberate though, but I think they should be loaded.

**IMPORTANT NOTE**: only `code.php` should be analysed, using eg `./vendor/bin/psalm --no-cache --debug --debug-by-line=true src/code.php`
